lgc/patch/PatchBufferOp.cpp

```diff
@@ -175,8 +175,16 @@ void PatchBufferOp::visitAtomicCmpXchgInst(AtomicCmpXchgInst &atomicCmpXchgInst)
 
     Value *const compareValue = atomicCmpXchgInst.getCompareOperand();
     Value *const newValue = atomicCmpXchgInst.getNewValOperand();
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 383129
+  // Old version of the code
     AtomicCmpXchgInst *const newAtomicCmpXchg =
         m_builder->CreateAtomicCmpXchg(atomicPointer, compareValue, newValue, successOrdering, failureOrdering);
+#else
+  // New version of the code (also handles unknown version, which we treat as latest)
+    AtomicCmpXchgInst *const newAtomicCmpXchg =
+        m_builder->CreateAtomicCmpXchg(atomicPointer, compareValue, newValue, MaybeAlign(),
```

Comment:
Do you think the MaybeAlign() here should be atomicCmpXchgInst.getAlign()? (and similarly for other cases where MaybeAlign() is used).
Note: Using MaybeAlign() gives the same functionality as before the change.

Reply:
If alignment could be improved, that should be a separate change, leaving this change to just cope with the upstream API change.